### ISSUE TITLE
guidance on the use of prov:qualifiedAttribution on Dataset would be welcome

### ISSUE THREAD

**Issue by NielsHoffmann** [SEMICeu/GeoDCAT-AP#143]
This is a remark coming from the GeoDCAT-AP 3.0.0 pilot.

On the Dataset there are properties defined for (a.o.) publisher, custodian and contact point. The same information can also be provided using the qualifiedAttribution property, which recommends the use of the [ResponsiblePartyRole](https://inspire.ec.europa.eu/metadata-codelist/ResponsiblePartyRole) codelist that contains those same values and more.

I appreciate the value of defining the most important roles as direct attributes. I do think it would be good to add an explanation to section 8.5 to clarify this and to clarify when to use wich option. 

Since all of the above mentioned properties are optional in GeoDCAT-AP it creates room for multiple implementations and therefore makes it more difficult to search for these properties. (essentially needing to check both options for the presence of a publisher for example)

**Comment by H-a-g-L** [comment]
Although I agree that [section 8.5](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#Attribution) could leave room to multiple implementations and would be in favour of editorial revision (even more so when considered together with [A.7.16](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#responsible-party-and-metadata-point-of-contact---dataset-responsible-party-and-metadata-point-of-contact)), I believe that [section 11. Agent Roles](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#agent-roles) explains how to implement this. Namely:

>The [[PROV-O](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#bib-prov-o)]-based approach is meant to give data providers an extension mechanism to specify, in a harmonised way, any kind of role, and, possibly, to attach additional information to the role and the associated Agent. As such, it MAY be used in combination with or to complement the property-based approach, but it MUST NOT replace it. For instance, it can be used to specify roles not covered by the agent role properties used in GeoDCAT-AP, and/or to specify additional information that cannot be expressed via a property (as the period during which an Agent covered a given role).

In other words, considering that all roles included in the [ResponsiblePartyRole](https://inspire.ec.europa.eu/metadata-codelist/ResponsiblePartyRole) codelist can and should be specified using the relevant properties (listed in [A.7.16](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#responsible-party-and-metadata-point-of-contact---dataset-responsible-party-and-metadata-point-of-contact)), in the INPIRE context, the `prov:qualifiedAttribution` metadata would only be useful either when the role of the attributed agent is not part of the INSPIRE list or when additional metadata regarding the Agent should be provided.

**Comment by hallinpihlatie** [comment]
A bit off the original question, but I just want to draw your attention to the work of ISO TC211. They are about to publish codelists online in a few months time. There is already information available online for comments. 

Would there be an idea to add for example the [CI Role Code](https://defs-dev.opengis.net/prez-hosted/catalogs/tc211:/collections/ns30:CI_RoleCode) to the list of the [Geo-DCAT-AP list of controlled vocabulaires to be used](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#controlled-vocabularies-to-be-used) in addition to the the [ResponsiblePartyRole](https://inspire.ec.europa.eu/metadata-codelist/ResponsiblePartyRole) one? Or would adding an additional code list just confuse more and complicating search?

**Comment by NielsHoffmann** [comment]
I would suggest adding a usage note to section 8.5.

The agent roles ```custodian, distributor, originator, principalInvestigator, processor, resourceProvider,``` and ```user``` should be described with the direct geodcatap properties. All other roles can be described using the ```prov:qualifiedAttribution``` statement.

**Comment by jescriu** [comment]
> I would suggest adding a usage note to section 8.5.
> 
> The agent roles custodian, distributor, originator, principalInvestigator, processor, resourceProvider, and user should be described with the direct geodcatap properties. All other roles can be described using the prov:qualifiedAttribution statement.

I would request the rest of ISO & GeoDCAT-AP pilot participants to raise if they use in their metadata any other agent roles different from those raised by Niels here. This would be important to check before proposing potential new values in GeoDCAT-AP.

**Comment by bertvannuffelen** [comment]
This is an topic with a lot of dimensions. The feedback below are generic considerations, to be considered as background knowledge for addressing the topic.

## 1. Considered from a data modeling perspective

This is about "similar semantical" relationships between two classes, for the simplicity I call them _Resource_ and _Agent._
Although the relationship is very similar, each relationship is unique in its own kind and expresses a different aspect how the two classes are related. Lets call the relationship _attribution_. And lets define two special kinds of this _attribution_: _builder_ and _architect_.

### Approach a): create subproperties

The relationship _attribution_ is encoded as a normal property connecting _Resource_ and _Agent_.

By definining _builder_ and _architect_ as subproperties of the property _attribution_ the specific semantics and the connection with the generic property are being set.

Advantages:
     - subproperties appear in the data specification
     - constraints: i.e. cardinalities, codelists can be imposed on them in the same way as normal properties
     - queries are trivial in case of retrieving specific properties (direct property), using the subproperty relationship also all involvements of _Agents_ in _Resources_ can be found.


Extension mechanism: 
     in case a new specific relationship is needed, the profile builder/ specification user needs to create a new subproperty. 
     If this needs to be shared with the public, a Linked Data publication with derefenceable URIs is needed for this new subproperty.
    There is no need to align with other stakeholders besides the profile community.

Disadvantages:
     - impossible to add extra information to a relationship without addition model change.
     - data model, visualisation can be cluttered with many optional properties. 

###  Approach b): use association class pattern

 The relationship _attribution_ is encoded as a class Attribution with two supporting properties connecting the 3 classes: _Resource_ -> _Attribution_ , and _Attribution_ -> _Agent_.
The association class _Attribution_  gets a supportive property _role_ with a codelist Role codelist.

The relationships _builder_ and _architect_ get to be a code in the Role codelist.

Advantages: 
   - allows to express additional values about how the relationship is being fullfilled. E.g. a weigth of the relationship. E.g. this resource is 50% build by Agent1, 30% by Agent2 and 20% by Agent3.
   
Extension mechanism
   - add a new code into the codelist. If this codelist is centrally managed then the addition should follow the rules of the codelist.
 
Disadvantages:
   - Additional constraints: cardinalities, codelists cannot be specified.
   - Queryies always involve a more complicated pattern. (2 supportive properties with a role with a specific value). Generic relationship querying can be done by ignore the role value.
   - If the role codelist has not been centralised, then each profile can use its own codelist with its own values. In that case it is possible that the same relationship e.g. is denoted by 2 distinct values codelist1:architect and codelist2:architect. 
   - Readers of the specification are required to consult another resource: the codelist Role, to know the possibilities of the relationships.


## 2. Considered from a master data perspective

If the data specification is the location to express the relationships between _Resource_ and _Agent_, the approach a) is the most natural. If one wants to discuss the role possibilities independent from the data specification, but in an abstract way, then the role codelist is probably the most natural approach. 

When the codelist is fixed then the management of the data specification is strongly bound with the management of the codelist. 
That is the reason why codelists are useally "flexibility"points in data specifications. They defer the discussion to another group, to the implemeters often. It is expected that codelists have another governance then the specification itself. 
,
In order to ensure that the data is shared in the same harmonised way, the codelist MUST be fixed to a single version. If not, there is a high risk on duplicates or variants of the same relationship. And thus a harmonisation activity must be imposed. This is the same harmonisation problem as with the subproperties created by distinct profiles. The Role codelist is thus not reducing the harmonisation effort, and may even create a more complicated harmonisation effort. 

**Conclusion**

Unless there is explicit need for support for the extra information (weights, etc) or the number of roles is very large and requires separate management the advantages of approach a) outweigth the need for approach b).

**Comment by NielsHoffmann** [comment]
> > I would suggest adding a usage note to section 8.5.
> > The agent roles custodian, distributor, originator, principalInvestigator, processor, resourceProvider, and user should be described with the direct geodcatap properties. All other roles can be described using the prov:qualifiedAttribution statement.
> 
> I would request the rest of ISO & GeoDCAT-AP pilot participants to raise if they use in their metadata any other agent roles different from those raised by Niels here. This would be important to check before proposing potential new values in GeoDCAT-AP.

To be clear. The roles I quoted here are the roles that are modeled directly in the geodcatap profile, other roles are to be specified via the prov:qualified attribution. 
So the request I am making is to merely clarify the use of roles in the documentation. I do not request any model changes.

**Comment by jescriu** [comment]
This issue was long discussed in the [Fourth Pilot meeting](https://github.com/INSPIRE-MIF/GeoDCAT-AP-pilot/tree/main/meetings/2025-04-28) on 28 April.

As preliminary conclusion, if the main roles are modelled as direct properties in the GeoDCAT-AP profile that takes precedence to the use of the `prov:qualifiedAttribution` construct. 

The suggestion from @NielsHoffmann is - To make this explicit in the GeoDCAT-AP 3 document. A separate discussion would be needed regarding the codelist specifying agent roles (and their values).

The issue is proposed for further discussion in the Fifth Pilot meeting, with the view of checking this preliminar conclusion and agreeing on a concrete action.

**Comment by NielsHoffmann** [comment]
Proposed text:

**Usage Note** 
    The agent roles ```custodian, distributor, originator, principalInvestigator, processor, resourceProvider```, and ```user``` should be described with the direct geodcatap properties. All other roles can be described using the prov:qualifiedAttribution statement.

**Comment by jakubklimek** [comment]
Sections [11. Agent Roles](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#agent-roles) and [A.7.16 Responsible party and metadata point of contact - *Dataset responsible party and *Metadata point of contact](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#responsible-party-and-metadata-point-of-contact---dataset-responsible-party-and-metadata-point-of-contact) now both state, that the PROV-O (`prov:qualifiedAttribution`) apporach is to be used only in addition to, and must not be used instead of, the direct property approach (also stated by @H-a-g-L  in https://github.com/SEMICeu/GeoDCAT-AP/issues/143#issuecomment-2804455135.

This seems to align with the proposition by @NielsHoffmann . However, the proposed usage note by @NielsHoffmann does not list the roles mapped to existing DCAT-AP properties - publisher, author, point of contact, and owner, suggesting that these should be described by `prov:qualifiedAttribution` instead, which contradicts the current wording.

Are these DCAT-AP properties to be added to the usage note, basically only emphasizing the current state, but also newly in section [8.5 Attribution](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#Attribution)?

**Comment by NielsHoffmann** [comment]
It was not my intention to introduce any contradiction into the usage note, quite the opposite in fact.
My bad for not including publisher, author, point of contact, and owner as properties that should be described with direct properties.

End result should be that for the reader of the document it is clear how to describe those properties. Since GeoDCAT-AP introduces a couple of extra direct properties (compared to DCAT-AP) it is important to explicitly clarify that to avoid any confusion.

**Comment by jakubklimek** [comment]
In order not to repeat the list of roles in multiple places, would it be satisfactory to just link to [11. Agent Roles](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#agent-roles) from the [8.5 Attribution](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#Attribution) usage note like so?

<img width="1498" height="155" alt="Image" src="https://github.com/user-attachments/assets/d4ea8a5e-46e0-4480-8e02-09ae9953fb3c" />

**Comment by NielsHoffmann** [comment]
I think so.
